protocols/zk-saas/src/network/mod.rs

```diff
@@ -244,10 +438,35 @@ fn handle_single_connection(
 
 #[derive(Serialize, Deserialize)]
 pub enum RegistryPacket {
-    Register { id: RegistantId, cert_der: Vec<u8> },
-    RegisterResponse { id: RegistantId, success: bool },
+    Register {
+        id: RegistantId,
+        cert_der: Vec<u8>,
+    },
+    RegisterResponse {
+        id: RegistantId,
+        success: bool,
+        king_registry_id: RegistantId,
+    },
     // A message for the substrate gadget
-    SubstrateGadgetMessage { payload: GadgetProtocolMessage },
+    SubstrateGadgetMessage {
```

Comment:
What about this warrants a `Substrate` prefix? I want to be very meticulous with things like this. If we don't these prefixes (or maybe generalized ones can be better `BlockchainGadgetMessage`) I'm curious to hear more ideas.